src/basic/parse-util.c

```diff
@@ -732,3 +733,33 @@ int parse_loadavg_fixed_point(const char *s, loadavg_t *ret) {
 
         return store_loadavg_fixed_point(i, f, ret);
 }
+
+static const char* const pcr_number_table[ALIASES_MAX] = {
+        [CORE_FM]     = "corefirmware",
+        [PLAT_CONFIG] = "platformconfig",
+        [EXT_CODE]    = "extcode",
+        [EXT_FM]      = "extfirmware",
+        [BOOT_LD]     = "bootloader",
+        [GPT_PT]      = "gptpartition",
+        [POWER_ST]    = "powerstate",
+        [SECURE_BT]   = "secureboot",
+        [LINUX_KN]    = "linuxkernel",
+        [IMA_ST]      = "imastate",
+        [SYSTEMD_KN]  = "systemdkernel",
+        [SYSTEMD_BT]  = "systemdboot",
+        [SYSTEMD_STB] = "systemdstub",
+        [SHIM_CT]     = "shimcerts",
+        [LUKS_KEY]    = "lukskey",
+};
+
+DEFINE_STRING_TABLE_LOOKUP_FROM_STRING(pcr_number, PcrNumber);
+
+int named_pcr(char *pcr, unsigned *n) {
```

Comment:
this is not needed anymore, the from_string function can be used directly

Reply:
> this is not needed anymore, the from_string function can be used directly

Alright sir.
Looks a little scary,. thinking about how I'll write the unit tests, But I'd rather write top notch code.

Thank you for all the guidance.